### ISSUE TITLE
Ruby bindings in MacOS Arm stuck on processing Heic images

### ISSUE THREAD

**Issue by miry** [libvips/libvips#4239]
## Bug report

**Describe the bug**
After upgrade vips package with homebrew on macos M1 the ruby process stuck on step write the file.

<img width="1675" alt="grafik" src="https://github.com/user-attachments/assets/48552bfa-26cd-4e7e-92f1-1c11e4328580">

Profile:

<img width="1226" alt="grafik" src="https://github.com/user-attachments/assets/c4f7e215-e12d-4199-802d-28bf8c601115">

```
$ brew info vips
==> vips: stable 8.16.0 (bottled)
Image processing library
https://github.com/libvips/libvips
Installed
/opt/homebrew/Cellar/vips/8.16.0 (85 files, 5.9MB) *
  Poured from bottle using the formulae.brew.sh API on 2024-11-02 at 11:11:47
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/v/vips.rb
License: LGPL-2.1-or-later
==> Dependencies
Build: gobject-introspection ✘, meson ✘, ninja ✘, pkg-config ✔
Required: cairo ✔, cfitsio ✔, cgif ✔, fftw ✔, fontconfig ✔, gettext ✔, glib ✔, highway ✔, imagemagick ✔, jpeg-xl ✔, libarchive ✔, libexif ✔, libheif ✔, libimagequant ✔, libmatio ✘, librsvg ✔, libspng ✔, libtiff ✔, little-cms2 ✔, mozjpeg ✔, openexr ✔, openjpeg ✔, openslide ✔, pango ✔, poppler ✔, webp ✔
==> Analytics
install: 9,339 (30 days), 22,889 (90 days), 81,146 (365 days)
install-on-request: 8,397 (30 days), 20,818 (90 days), 73,494 (365 days)
build-error: 1 (30 days)

$ file test.heic 
test.heic: ISO Media, HEIF Image HEVC Main or Main Still Picture Profile

$ gem install ruby-vips
Successfully installed ruby-vips-2.2.2

$ irb
irb(main):001> require "ruby-vips"
=> true
irb(main):002> Vips::VERSION
=> "2.2.2"
irb(main):003> image = Vips::Image.new_from_file("test.heic")
=> #<Image 1x1 uchar, 3 bands, srgb>
irb(main):004> image.write_to_file("new_test.heic")

<stuck>
```

<details><summary>lsof</summary>

<pre>

COMMAND   PID USER   FD     TYPE             DEVICE SIZE/OFF      NODE NAME
ruby    72751 miry  cwd      DIR               1,18     1888 183399978 /Users/miry/src/project
ruby    72751 miry  txt      REG               1,18    34200 219997391 /Users/miry/.rbenv/versions/3.3.5/bin/ruby
ruby    72751 miry  txt      REG               1,18    64604 228181526 /Library/Preferences/Logging/.plist-cache.OPtXb3HB
ruby    72751 miry  txt      REG               1,18   265824 225869831 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/ffi-1.16.3/lib/ffi_c.bundle
ruby    72751 miry  txt      REG               1,18   452816 169424466 /opt/homebrew/Cellar/gmp/6.3.0/lib/libgmp.10.dylib
ruby    72751 miry  txt      REG               1,18    50576 219997444 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/enc/encdb.bundle
ruby    72751 miry  txt      REG               1,18    50352 219997459 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/enc/trans/transdb.bundle
ruby    72751 miry  txt      REG               1,18    53424 219997493 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/monitor.bundle
ruby    72751 miry  txt      REG               1,18    60176 225869143 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/io-console-0.7.2/lib/io/console.bundle
ruby    72751 miry  txt      REG               1,18    57128 219997405 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/etc.bundle
ruby    72751 miry  txt      REG               1,18    59475 228654309 /opt/homebrew/Cellar/vips/8.16.0/share/locale/de/LC_MESSAGES/vips8.16.mo
ruby    72751 miry  txt      REG               1,18    51176 219997415 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/io/wait.bundle
ruby    72751 miry  txt      REG               1,18    52320 209267048 /opt/homebrew/Cellar/cgif/0.4.1/lib/libcgif.0.dylib
ruby    72751 miry  txt      REG               1,18    53376 198729409 /opt/homebrew/Cellar/webp/1.4.0/lib/libwebpdemux.2.0.15.dylib
ruby    72751 miry  txt      REG               1,18    54224 188350533 /opt/homebrew/Cellar/libxdmcp/1.1.5/lib/libXdmcp.6.dylib
ruby    72751 miry  txt      REG               1,18    56784 223129719 /opt/homebrew/Cellar/jpeg-xl/0.11.0_1/lib/libjxl_threads.0.11.0.dylib
ruby    72751 miry  txt      REG               1,18   260688 219997418 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/ripper.bundle
ruby    72751 miry  txt      REG               1,18  6270944 219997393 /Users/miry/.rbenv/versions/3.3.5/lib/libruby.3.3.dylib
ruby    72751 miry  txt      REG               1,18    53312 132268967 /opt/homebrew/Cellar/libxau/1.0.11/lib/libXau.6.0.0.dylib
ruby    72751 miry  txt      REG               1,18    91504 219997419 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/fiddle.bundle
ruby    72751 miry  txt      REG               1,18    78896 219997413 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/pathname.bundle
ruby    72751 miry  txt      REG               1,18    80368 219997404 /Users/miry/.rbenv/versions/3.3.5/lib/ruby/3.3.0/arm64-darwin23/stringio.bundle
ruby    72751 miry  txt      REG               1,18    71264 226001451 /opt/homebrew/Cellar/glib/2.82.2/lib/libgmodule-2.0.0.dylib
ruby    72751 miry  txt      REG               1,18    91360 183412370 /opt/homebrew/Cellar/libspng/0.7.4/lib/libspng.0.dylib
ruby    72751 miry  txt      REG               1,18    73552 198729410 /opt/homebrew/Cellar/webp/1.4.0/lib/libwebpmux.3.1.0.dylib
ruby    72751 miry  txt      REG               1,18   162656 187420751 /opt/homebrew/Cellar/gettext/0.22.5/lib/libintl.8.dylib
ruby    72751 miry  txt      REG               1,18    96928 206573851 /opt/homebrew/Cellar/highway/1.2.0/lib/libhwy.1.2.0.dylib
ruby    72751 miry  txt      REG               1,18   154944 208303493 /opt/homebrew/Cellar/pango/1.54.0/lib/libpangocairo-1.0.0.dylib
ruby    72751 miry  txt      REG               1,18   372448 226001452 /opt/homebrew/Cellar/glib/2.82.2/lib/libgobject-2.0.0.dylib
ruby    72751 miry  txt      REG               1,18   264112 183411150 /opt/homebrew/Cellar/libexif/0.6.24/lib/libexif.12.dylib
ruby    72751 miry  txt      REG               1,18   353664 198729407 /opt/homebrew/Cellar/webp/1.4.0/lib/libwebp.7.1.9.dylib
ruby    72751 miry  txt      REG               1,18   687968 183410821 /opt/homebrew/Cellar/fftw/3.3.10_1/lib/libfftw3.3.dylib
ruby    72751 miry  txt      REG               1,18  2308080 228654327 /opt/homebrew/Cellar/vips/8.16.0/lib/libvips.42.dylib
ruby    72751 miry  txt      REG               1,18  1201152 226001450 /opt/homebrew/Cellar/glib/2.82.2/lib/libglib-2.0.0.dylib
ruby    72751 miry  txt      REG               1,18  1829616 226001448 /opt/homebrew/Cellar/glib/2.82.2/lib/libgio-2.0.0.dylib
ruby    72751 miry  txt      REG               1,18   704384 224691954 /opt/homebrew/Cellar/libarchive/3.7.7/lib/libarchive.13.dylib
ruby    72751 miry  txt      REG               1,18   661712 215913780 /opt/homebrew/Cellar/libimagequant/4.3.3/lib/libimagequant.0.4.0.dylib
ruby    72751 miry  txt      REG               1,18   408400 183412509 /opt/homebrew/Cellar/mozjpeg/4.1.5/lib/libjpeg.62.3.0.dylib
ruby    72751 miry  txt      REG               1,18    69440  74893451 /opt/homebrew/Cellar/libb2/0.98.1/lib/libb2.1.dylib
ruby    72751 miry  txt      REG               1,18  1186176 209266991 /opt/homebrew/Cellar/cfitsio/4.4.1/lib/libcfitsio.10.4.4.1.dylib
ruby    72751 miry  txt      REG               1,18   139056 208303494 /opt/homebrew/Cellar/pango/1.54.0/lib/libpangoft2-1.0.0.dylib
ruby    72751 miry  txt      REG               1,18    69872 198729406 /opt/homebrew/Cellar/webp/1.4.0/lib/libsharpyuv.0.1.0.dylib
ruby    72751 miry  txt      REG               1,18    89616 135470664 /opt/homebrew/Cellar/libxrender/0.9.11/lib/libXrender.1.dylib
ruby    72751 miry  txt      REG               1,18    70432 200095248 /opt/homebrew/Cellar/libxcb/1.17.0/lib/libxcb-shm.0.0.0.dylib
ruby    72751 miry  txt      REG               1,18   411552 208303492 /opt/homebrew/Cellar/pango/1.54.0/lib/libpango-1.0.0.dylib
ruby    72751 miry  txt      REG               1,18   185136 223102280 /opt/homebrew/Cellar/xz/5.6.3/lib/liblzma.5.dylib
ruby    72751 miry  txt      REG               1,18   305760 179105948 /opt/homebrew/Cellar/fontconfig/2.15.0/lib/libfontconfig.1.dylib
ruby    72751 miry  txt      REG               1,18   176960 217645208 /opt/homebrew/Cellar/lz4/1.10.0/lib/liblz4.1.10.0.dylib
ruby    72751 miry  txt      REG               1,18   171920 221928102 /opt/homebrew/Cellar/fribidi/1.0.16/lib/libfribidi.0.dylib
ruby    72751 miry  txt      REG               1,18   138640  74889971 /opt/homebrew/Cellar/graphite2/1.3.14/lib/libgraphite2.3.2.1.dylib
ruby    72751 miry  txt      REG               1,18  1178384 219731640 /opt/homebrew/Cellar/cairo/1.18.2/lib/libcairo.2.dylib
ruby    72751 miry  txt      REG               1,18   543120 221335861 /opt/homebrew/Cellar/libtiff/4.7.0/lib/libtiff.6.dylib
ruby    72751 miry  txt      REG               1,18   380304 222430810 /opt/homebrew/Cellar/libmatio/1.5.27_1/lib/libmatio.13.dylib
ruby    72751 miry  txt      REG               1,18   357184 179243210 /opt/homebrew/Cellar/little-cms2/2.16/lib/liblcms2.2.dylib
ruby    72751 miry  txt      REG               1,18   325856 188141656 /opt/homebrew/Cellar/openjpeg/2.5.2/lib/libopenjp2.2.5.2.dylib
ruby    72751 miry  txt      REG               1,18   545424 208623784 /opt/homebrew/Cellar/pcre2/10.44/lib/libpcre2-8.0.dylib
ruby    72751 miry  txt      REG               1,18   115216 187421149 /opt/homebrew/Cellar/libxext/1.3.6/lib/libXext.6.dylib
ruby    72751 miry  txt      REG               1,18  1045680 224258037 /opt/homebrew/Cellar/openexr/3.3.1/lib/libOpenEXR-3_3.32.3.3.1.dylib
ruby    72751 miry  txt      REG               1,18   105568 200095243 /opt/homebrew/Cellar/libxcb/1.17.0/lib/libxcb-render.0.0.0.dylib
ruby    72751 miry  txt      REG               1,18    75936 219731638 /opt/homebrew/Cellar/cairo/1.18.2/lib/libcairo-gobject.2.dylib
ruby    72751 miry  txt      REG               1,18   209568 220481113 /opt/homebrew/Cellar/libpng/1.6.44/lib/libpng16.16.dylib
ruby    72751 miry  txt      REG               1,18   194768 200095263 /opt/homebrew/Cellar/libxcb/1.17.0/lib/libxcb.1.1.0.dylib
ruby    72751 miry  txt      REG               1,18   669440 197750781 /opt/homebrew/Cellar/zstd/1.5.6/lib/libzstd.1.5.6.dylib
ruby    72751 miry  txt      REG               1,18   641712 217831913 /opt/homebrew/Cellar/freetype/2.13.3/lib/libfreetype.6.dylib
ruby    72751 miry  txt      REG               1,18    71344 193946945 /opt/homebrew/Cellar/libaec/1.1.3/lib/libsz.2.0.1.dylib
ruby    72751 miry  txt      REG               1,18    69744 224258036 /opt/homebrew/Cellar/openexr/3.3.1/lib/libIlmThread-3_3.32.3.3.1.dylib
ruby    72751 miry  txt      REG               1,18   221376 205397260 /opt/homebrew/Cellar/gdk-pixbuf/2.42.12/lib/libgdk_pixbuf-2.0.0.dylib
ruby    72751 miry  txt      REG               1,18  1173920 223859556 /opt/homebrew/Cellar/harfbuzz/10.0.1_1/lib/libharfbuzz.0.dylib
ruby    72751 miry  txt      REG               1,18  6644816 220481355 /opt/homebrew/Cellar/librsvg/2.58.4/lib/librsvg-2.2.dylib
ruby    72751 miry  txt      REG               1,18   373728 129091974 /opt/homebrew/Cellar/pixman/0.42.2/lib/libpixman-1.0.42.2.dylib
ruby    72751 miry  txt      REG               1,18   473216 221213428 /opt/homebrew/Cellar/jpeg-turbo/3.0.4/lib/libjpeg.8.3.2.dylib
ruby    72751 miry  txt      REG               1,18   105104 224257362 /opt/homebrew/Cellar/libdeflate/1.22/lib/libdeflate.0.dylib
ruby    72751 miry  txt      REG               1,18  1025536 215642255 /opt/homebrew/Cellar/libx11/1.8.10/lib/libX11.6.dylib
ruby    72751 miry  txt      REG               1,18   102272 228654328 /opt/homebrew/Cellar/vips/8.16.0/lib/vips-modules-8.16/vips-heif.dylib
ruby    72751 miry  txt      REG               1,18    99760 228654330 /opt/homebrew/Cellar/vips/8.16.0/lib/vips-modules-8.16/vips-magick.dylib
ruby    72751 miry  txt      REG               1,18   110224 219728487 /opt/homebrew/Cellar/liblqr/0.4.3/lib/liblqr-1.0.dylib
ruby    72751 miry  txt      REG               1,18    93792 221928346 /opt/homebrew/Cellar/libtool/2.5.3/lib/libltdl.7.dylib
ruby    72751 miry  txt      REG               1,18   571168 224258035 /opt/homebrew/Cellar/openexr/3.3.1/lib/libIex-3_3.32.3.3.1.dylib
ruby    72751 miry  txt      REG               1,18   357824 220369410 /opt/homebrew/Cellar/imath/3.1.12/lib/libImath-3_1.29.11.0.dylib
ruby    72751 miry  txt      REG               1,18    93248 228654332 /opt/homebrew/Cellar/vips/8.16.0/lib/vips-modules-8.16/vips-poppler.dylib
ruby    72751 miry  txt      REG               1,18    69920 226668026 /opt/homebrew/Cellar/nspr/4.36/lib/libplds4.dylib
ruby    72751 miry  txt      REG               1,18    71056 226668025 /opt/homebrew/Cellar/nspr/4.36/lib/libplc4.dylib
ruby    72751 miry  txt      REG               1,18    92480 228654331 /opt/homebrew/Cellar/vips/8.16.0/lib/vips-modules-8.16/vips-openslide.dylib
ruby    72751 miry  txt      REG               1,18   438800 178433519 /opt/homebrew/Cellar/libde265/1.0.15/lib/libde265.0.dylib
ruby    72751 miry  txt      REG               1,18   889888 227945916 /opt/homebrew/Cellar/libheif/1.18.2_1/lib/libheif.1.18.2.dylib
ruby    72751 miry  txt      REG               1,18   116896 215907958 /opt/homebrew/Cellar/libassuan/3.0.1/lib/libassuan.9.dylib
ruby    72751 miry  txt      REG               1,18  3534160 222428808 /opt/homebrew/Cellar/hdf5/1.14.4.3/lib/libhdf5.310.4.0.dylib
ruby    72751 miry  txt      REG               1,18  2270896 224258038 /opt/homebrew/Cellar/openexr/3.3.1/lib/libOpenEXRCore-3_3.32.3.3.1.dylib
ruby    72751 miry  txt      REG               1,18   223072 226668343 /opt/homebrew/Cellar/nss/3.106/lib/libnssutil3.dylib
ruby    72751 miry  txt      REG               1,18   219728 226668344 /opt/homebrew/Cellar/nss/3.106/lib/libsmime3.dylib
ruby    72751 miry  txt      REG               1,18   180208 209839407 /opt/homebrew/Cellar/libgpg-error/1.50/lib/libgpg-error.0.dylib
ruby    72751 miry  txt      REG               1,18    98976 228654329 /opt/homebrew/Cellar/vips/8.16.0/lib/vips-modules-8.16/vips-jxl.dylib
ruby    72751 miry  txt      REG               1,18   747776 180279854 /opt/homebrew/Cellar/libvmaf/3.0.0/lib/libvmaf.3.dylib
ruby    72751 miry  txt      REG               1,18   735584 225151989 /opt/homebrew/Cellar/libomp/19.1.2/lib/libomp.dylib
ruby    72751 miry  txt      REG               1,18   519056 215914828 /opt/homebrew/Cellar/poppler/24.04.0_1/lib/libpoppler-glib.8.26.0.dylib
ruby    72751 miry  txt      REG               1,18   414688 215908887 /opt/homebrew/Cellar/gpgme/1.23.2_2/lib/libgpgmepp.6.dylib
ruby    72751 miry  txt      REG               1,18   240112 226668024 /opt/homebrew/Cellar/nspr/4.36/lib/libnspr4.dylib
ruby    72751 miry  txt      REG               1,18   111504 223129718 /opt/homebrew/Cellar/jpeg-xl/0.11.0_1/lib/libjxl_cms.0.11.0.dylib
ruby    72751 miry  txt      REG               1,18   383568 226668346 /opt/homebrew/Cellar/nss/3.106/lib/libssl3.dylib
ruby    72751 miry  txt      REG               1,18    88016 163170959 /opt/homebrew/Cellar/brotli/1.1.0/lib/libbrotlidec.1.1.0.dylib
ruby    72751 miry  txt      REG               1,18   144654 226001177 /opt/homebrew/Cellar/glib/2.82.2/share/locale/de/LC_MESSAGES/glib20.mo
ruby    72751 miry  txt      REG               1,18   168128 163170958 /opt/homebrew/Cellar/brotli/1.1.0/lib/libbrotlicommon.1.1.0.dylib
ruby    72751 miry  txt      REG               1,18  4743424 227943836 /opt/homebrew/Cellar/x265/4.0/lib/libx265.212.dylib
ruby    72751 miry  txt      REG               1,18  4034736 219730265 /opt/homebrew/Cellar/aom/3.10.0/lib/libaom.3.10.0.dylib
ruby    72751 miry  txt      REG               1,18  1961424 223605021 /opt/homebrew/Cellar/imagemagick/7.1.1-39/lib/libMagickCore-7.Q16HDRI.10.dylib
ruby    72751 miry  txt      REG               1,18  1179520 226668339 /opt/homebrew/Cellar/nss/3.106/lib/libnss3.dylib
ruby    72751 miry  txt      REG               1,18   332208 215908886 /opt/homebrew/Cellar/gpgme/1.23.2_2/lib/libgpgme.11.dylib
ruby    72751 miry  txt      REG               1,18   263056 183412695 /opt/homebrew/Cellar/openslide/4.0.0/lib/libopenslide.1.dylib
ruby    72751 miry  txt      REG               1,18   542992 198949882 /opt/homebrew/Cellar/libdicom/1.1.0/lib/libdicom.1.dylib
ruby    72751 miry  txt      REG               1,18   117156 183411129 /opt/homebrew/Cellar/libexif/0.6.24/share/locale/de/LC_MESSAGES/libexif-12.mo
ruby    72751 miry  txt      REG               1,18  3203648 215914829 /opt/homebrew/Cellar/poppler/24.04.0_1/lib/libpoppler.136.0.0.dylib
ruby    72751 miry  txt      REG               1,18  1093440 224227220 /opt/homebrew/Cellar/libxml2/2.13.4_2/lib/libxml2.2.dylib
ruby    72751 miry  txt      REG               1,18   639312 163170960 /opt/homebrew/Cellar/brotli/1.1.0/lib/libbrotlienc.1.1.0.dylib
ruby    72751 miry  txt      REG               1,18  3154528 223794828 /opt/homebrew/Cellar/icu4c@75/75.1/lib/libicui18n.75.1.dylib
ruby    72751 miry  txt      REG               1,18  1814528 223794832 /opt/homebrew/Cellar/icu4c@75/75.1/lib/libicuuc.75.1.dylib
ruby    72751 miry  txt      REG               1,18  1942032 223129717 /opt/homebrew/Cellar/jpeg-xl/0.11.0_1/lib/libjxl.0.11.0.dylib
ruby    72751 miry  txt      REG               1,18 30994896 223794827 /opt/homebrew/Cellar/icu4c@75/75.1/lib/libicudata.75.1.dylib
ruby    72751 miry    0u     CHR               16,3 0t156337      1405 /dev/ttys003
ruby    72751 miry    1u     CHR               16,3 0t156337      1405 /dev/ttys003
ruby    72751 miry    2u     CHR               16,3 0t156337      1405 /dev/ttys003
ruby    72751 miry    3     PIPE 0x1ee5999d3fd2a42d    16384           ->0x671b1acab9aa0c5d
ruby    72751 miry    4     PIPE 0x671b1acab9aa0c5d    16384           ->0x1ee5999d3fd2a42d
ruby    72751 miry    5u  KQUEUE                                       count=0, state=0xa
ruby    72751 miry    6u     REG               1,18        0 228684805 /Users/miry/src/project/new_test.heic

</pre>
</details>

**Comment by jcupitt** [comment]
Hello @miry,

It looks like there´s a problem with the current vips package on homebrew -- I think it needs to be built without module support.

```
11:27 $ brew update && brew upgrade
==> Updating Homebrew...
Already up-to-date.
✔ ~ 
11:27 $ vips --version
vips-8.16.0
✔ ~ 
11:27 $ vips --vips-config | grep heif
HEIC/AVIF load/save with libheif: true (dynamic module: true)
✔ ~ 
11:27 $ vips heifload
vips: unknown action "heifload"
✘-1 ~ 
```

I´ll make a PR.

**Comment by kleisauke** [comment]
Unfortunately, using `-Dmodules=disabled` disables all module loading, so custom modules placed in `$(brew --prefix vips)/lib/vips-modules-x.y` will no longer work. @jcupitt Does running with `VIPS_INFO=1` display anything noteworthy?

**Comment by jcupitt** [comment]
Oop! Thanks Kleis, I had a stray `VIPSHOME` env var which was blocking module load.

There's another issue now:

```
12:38 $ vips copy k2.jpg x.avif
✔ ~/pics 
12:39 $ vips copy k2.jpg x.heic
Segmentation fault: 11
✘-SEGV ~/pics 
```

**Comment by kleisauke** [comment]
The stack trace in the initial post suggests an issue with x265. Perhaps updating to libheif v1.19.1, as proposed in PR https://github.com/Homebrew/homebrew-core/pull/196350, could resolve it?

**Comment by jcupitt** [comment]
lldb says:

```
(lldb) bt
* thread #2, name = 'libvips worker', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
  * frame #0: 0x0000000103483390 libx265.212.dylib`x265::Encoder::encode(x265_picture const*, x265_picture**) + 3436
    frame #1: 0x000000010348f58c libx265.212.dylib`x265_encoder_encode + 84
    frame #2: 0x000000010019d650 libheif.1.18.2.dylib`x265_get_compressed_data(void*, unsigned char**, int*, heif_encoded_data_type*) + 244
    frame #3: 0x000000010015461c libheif.1.18.2.dylib`HeifContext::encode_image_as_hevc(std::__1::shared_ptr<HeifPixelImage> const&, heif_encoder*, heif_encoding_options const&, heif_image_input_class, std::__1::shared_ptr<HeifContext::Image>&) + 832
    frame #4: 0x00000001001540c4 libheif.1.18.2.dylib`HeifContext::encode_image(std::__1::shared_ptr<HeifPixelImage> const&, heif_encoder*, heif_encoding_options const&, heif_image_input_class, std::__1::shared_ptr<HeifContext::Image>&) + 176
    frame #5: 0x0000000100173fd0 libheif.1.18.2.dylib`heif_context_encode_image + 160
    frame #6: 0x0000000100039d20 vips-heif.dylib`vips_foreign_save_heif_write_page(heif=0x0000600000bb4a00, page=0) at heifsave.c:321:11
    frame #7: 0x0000000100039780 vips-heif.dylib`vips_foreign_save_heif_write_block(region=0x00006000000b04d0, area=0x00006000000b0430, a=0x0000600000bb4a00) at heifsave.c:476:8
```

Where:

```
12:43 $ brew info x265
==> x265: stable 4.0 (bottled), HEAD
H.265/HEVC encoder
https://bitbucket.org/multicoreware/x265_git
Installed
/opt/homebrew/Cellar/x265/4.0_1 (12 files, 10.8MB) *
```

I'll try 4.0 on ubuntu (they are on 3.5 right now).

**Comment by jcupitt** [comment]
Ahh! Yes, they are very behind on libheif:

```
12:43 $ brew info libheif
==> libheif: stable 1.18.2 (bottled)
```

**Comment by kleisauke** [comment]
I think commit https://github.com/strukturag/libheif/commit/4d9d27e0b2580de23f7c620aeed5553554e19f95 fixes this. See:
https://bitbucket.org/multicoreware/x265_git/issues/952/crash-in-libheif-tests

**Comment by jcupitt** [comment]
Yes, that sounds like it. I'll try updating libheif to 1.19.1.

**Comment by jcupitt** [comment]
Ah I see you're on the libheif 1.19.1 issue heh.

@miry This looks like a homebrew issue and should be resolved by this PR:

https://github.com/Homebrew/homebrew-core/pull/196350

I'll close this issue, please reopen if you still have problems once that update lands.

**Comment by miry** [comment]
Thank you @jcupitt and @kleisauke

**Comment by miry** [comment]
The fix https://github.com/Homebrew/homebrew-core/pull/196350 merged and it works.

**Comment by jcupitt** [comment]
Great! Thanks for testing and confirming.
